mainnet_solver_rewards/service_fee_query_4017925.sql

```diff
@@ -0,0 +1,175 @@
+  WHERE time > CAST('2024-08-20 00:00:00' AS timestamp) -- CIP-48 starts bonding pool timer at midnight UTC on 20/08/24
+),
+
+initial_vouches AS (
+  SELECT RANK() OVER (
+      PARTITION BY solver, bondingPool, sender
+      ORDER BY evt_block_number ASC, evt_index ASC 
+    ) AS rk,
+    evt_block_number,
+    evt_index,
+    solver,
+    cowRewardTarget,
+    bondingPool,
+    sender,
+    True AS active
+  FROM cow_protocol_ethereum.VouchRegister_evt_Vouch
+  WHERE evt_block_number <= (SELECT * FROM first_event_after_timestamp)
+    AND bondingPool IN (SELECT pool FROM bonding_pools)
+    AND sender IN (SELECT initial_funder FROM bonding_pools)
+),
+
+joined_on_data AS (
+  SELECT
+    iv.solver,
+    iv.cowRewardTarget AS reward_target,
+    iv.bondingPool AS pool,
+    iv.evt_block_number,
+    iv.evt_index,
+    iv.rk,
+    True AS active
+  FROM initial_vouches iv
+  WHERE iv.rk = 1
+),
+
+latest_vouches AS (
+  SELECT RANK() OVER (
+      PARTITION BY solver, bondingPool, sender
+      ORDER BY evt_block_number DESC, evt_index DESC 
+    ) AS rk,
+    evt_block_number,
+    evt_index,
+    solver,
+    cowRewardTarget,
+    bondingPool,
+    sender,
+    CASE WHEN event_type = 'Vouch' THEN True ELSE False END AS active
+  FROM (
+      SELECT 
+        evt_block_number,
+        evt_index,
+        solver,
+        cowRewardTarget,
+        bondingPool,
+        sender,
+        'Vouch' AS event_type
+      FROM cow_protocol_ethereum.VouchRegister_evt_Vouch
+      WHERE evt_block_number <= (SELECT * FROM first_event_after_timestamp)
+        AND bondingPool IN (SELECT pool FROM bonding_pools)
+        AND sender IN (SELECT initial_funder FROM bonding_pools)
+        
+      UNION
+      
+      SELECT 
+        evt_block_number,
+        evt_index,
+        solver,
+        NULL AS cowRewardTarget,  -- Invalidation does not have a reward target
+        bondingPool,
+        sender,
+        'InvalidateVouch' AS event_type
+      FROM cow_protocol_ethereum.VouchRegister_evt_InvalidateVouch
+      WHERE evt_block_number <= (SELECT * FROM first_event_after_timestamp)
+        AND bondingPool IN (SELECT pool FROM bonding_pools)
+        AND sender IN (SELECT initial_funder FROM bonding_pools)
+  ) AS unioned_events
+),
+
+valid_vouches AS (
+  SELECT
+    lv.solver,
+    lv.cowRewardTarget AS reward_target,
+    lv.bondingPool AS pool
+  FROM latest_vouches lv
+  WHERE lv.rk = 1 AND lv.active = TRUE
+),
+
+joined_on AS (
+  SELECT
+    jd.solver,
+    jd.reward_target,
+    jd.pool,
+    bp.name AS pool_name,
+    b.time AS joined_on
+  FROM joined_on_data jd
+  JOIN ethereum.blocks b
+    ON b.number = jd.evt_block_number
+  JOIN bonding_pools bp
+    ON jd.pool = bp.pool
+),
+
+named_results AS (
+  SELECT
+    jd.solver,
+    CONCAT(environment, '-', s.name) AS solver_name,
+    jd.pool_name,
+    jd.pool,
+    jd.joined_on,
+    date_diff('day', date(jd.joined_on), date(NOW())) AS days_in_pool
```

Comment:
`NOW()` should probably be changed to a `cutoff_time` parameter which would chosen as `start_time`.